sklearn/decomposition/nmf.py

```diff
@@ -1049,7 +1046,6 @@ def fit_transform(self, X, y=None, W=None, H=None):
 
         self.n_components_ = H.shape[0]
         self.components_ = H
-        self.n_iter_ = n_iter_
```

Comment:
You should not have removed this. We need it in `fit_transform`, not in `transform`